src/BioSequences.jl

```diff
@@ -227,4 +227,46 @@ include("search/ApproxSearchQuery.jl")
 include("search/re.jl")
 include("search/pwm.jl")
 
+
+
+struct Search{Q,I}
+    query::Q
+    itr::I
+    start::Integer
+    stop::Integer
+    overlap::Bool
+end
+
+const default_overlap = true
+
+search(query, itr, start=firstindex(itr); overlap = default_overlap, stop = lastindex(itr)) = Search(query, itr, start, stop, overlap)
+
+function Base.iterate(itr::Search, state=itr.start)
+    val = findnext(itr.query, itr.itr, state)
+    val === nothing && return nothing
+    last(val) > itr.stop && return nothing
+    state = itr.overlap ? first(val) + 1 : last(val) + 1
+    return val, state
+end
+
+const HasRangeEltype = Union{<:ExactSearchQuery, <:ApproximateSearchQuery, <:Regex}
+
+Base.eltype(::Type{Search{Q,I}}) where {Q<:HasRangeEltype,I} = UnitRange{Int}
+Base.IteratorEltype(::Type{Search{Q,I}}) where {Q<:HasRangeEltype,I} = Base.IteratorEltype(UnitRange{Int})
+
+Base.eltype(::Type{<:Search}) = Int
+Base.IteratorEltype(::Type{<:Search}) = Base.IteratorEltype(Int)
+
+Base.IteratorSize(::Type{<:Search}) = Base.SizeUnknown()
```

Comment:
```suggestion
Base.eltype(::Type{Search{<:Q}}) where {Q<:HasRangeEltype} = UnitRange{Int}
Base.eltype(::Type{<:Search}) = Int
Base.IteratorSize(::Type{<:Search}) = Base.SizeUnknown()
```

Reply:
`Base.IteratorSize` just needs to be `HasEltype()`, but this is already the default.

Reply:
@jakobnissen, do we need the `<:Q`?
```julia
Base.eltype(::Type{<:Search{<:Q}}) where {Q<:HasRangeEltype} = UnitRange{Int}
vs
Base.eltype(::Type{<:Search{Q}}) where {Q<:HasRangeEltype} = UnitRange{Int}
```

Reply:
Ah no, you are right, we don't